### PR TITLE
[dv/uvmdvgen] update dvsim and remove Makefile

### PR DESCRIPTION
**Pull request by cindychip**
From PR #1855, when generate dv using uvmdvgen, there are some old
comments and code still using make.
This PR updates the uvmdvgen templates by:
1. Remove the Makefile.tpl
2. Update command from make to dvsim

Signed-off-by: Cindy Chen <chencindy@google.com>

### CODE REVIEW

File: util/uvmdvgen/index.md.tpl [hunk 1]
```diff
@@ -120,7 +120,7 @@ Please take a look at the link for detailed information on the usage, capabiliti
 Here's how to run a basic sanity test:
 ```console
 $ cd hw/ip/${name}/dv
-$ make TEST_NAME=${name}_sanity
+$ ../../../util/dvsim/dvsim.py ${name}_sim_cfg.hjson -i ${name}_sanity
```
Comment (sriyerg) [inline util/uvmdvgen/index.md.tpl]: Probably better to ref the command from $REPO_TOP.

File: util/uvmdvgen/gen_env.py [hunk 2]
```diff
@@ -55,17 +55,21 @@ def gen_env(name, is_cip, has_ral, has_interrupts, has_alerts, env_agents,
         src_suffix = tup[3]
 
         # Skip Makefile
-        if src == 'Makefile' and not add_makefile: continue
+        if src == 'Makefile' and not add_makefile:
+            continue
```
Comment (sriyerg) [inline util/uvmdvgen/gen_env.py:59]: Please remove this as well - seems unnecessary now.

Comment (sriyerg) [reply util/uvmdvgen/gen_env.py:59]: You can remove `add_makefile` arg in the function and remove the corresponding switch in `uvmdvgen.py` as well.

**Review by sriyerg — COMMENTED** [review]
> From PR #1855, when generate dv using uvmdvgen, there are some old
> comments and code still using make.
> This PR updates the uvmdvgen templates by:
> 
> 1. Remove the Makefile.tpl
> 2. Update command from make to dvsim
> 
> Signed-off-by: Cindy Chen [chencindy@google.com](mailto:chencindy@google.com)

Actually, the Makefile is not generated by default anymore. User has to pass `--add-makefile` switch explicitly to `uvmdvgen.py`. But, I fully support the complete removal of Makefile, since we are not supporting it anymore.

File: util/uvmdvgen/index.md.tpl (hunk 1 above)
Comment (cindychip) [reply util/uvmdvgen/index.md.tpl]: Sounds good, thanks Sri I will update that.

**Review by sriyerg — APPROVED** [review]
LG, thanks!

**Comment by cindychip** [comment]
This PR solves part of issue : https://github.com/lowRISC/opentitan/issues/3306

**Pull request merged**